src/test/java/org/kainos/ea/dao/JobRoleDaoTest.java

```diff
@@ -125,4 +126,44 @@ public void updateJobRole_shouldThrowFailedToUpdateJobRoleException_whenCantUpda
         assertThrows(FailedToUpdateJobRoleException.class,
                 () -> jobRoleDao.updateJobRole(1,jobRoleRequest));
     }
+
+    @Test
+    public void createJobRole_shouldReturnId_whenValidJobRole() throws SQLException, FailedToCreateJobRoleRequestException {
+        CreateJobRoleRequest jobRoleRequest = new CreateJobRoleRequest("testName",
+                                                                            "testJobSpec",
+                                                                            1,
+                                                                            1,
+                                                                            "testResponsibilities",
+                                                                            "https://kainossoftwareltd.sharepoint.com/:b:/r/people/Job%20Specifications/Engineering/Job%20profile%20-%20Software%20Engineer%20(Trainee).pdf?csf=1&web=1&e=nQzHld"
+        );
+        Connection connectionMock = mock(Connection.class);
+        Mockito.when(databaseConnector.getConnection()).thenReturn(connectionMock);
+        PreparedStatement mockPreparedStatement = mock(PreparedStatement.class);
+        Mockito.when(connectionMock.prepareStatement(any(String.class), eq(Statement.RETURN_GENERATED_KEYS)))
+                .thenReturn(mockPreparedStatement);
+
+
+        ResultSet resultSet = mock(ResultSet.class);
+        Mockito.when(mockPreparedStatement.getGeneratedKeys()).thenReturn(resultSet);
+
+        Mockito.when(resultSet.next()).thenReturn(true);
+        Mockito.when(resultSet.getInt(anyInt())).thenReturn(1);
+        int id = jobRoleDao.createJobRole(jobRoleRequest);
+        Assertions.assertEquals(1,id);
+    }
+
+    @Test
+    public void createJobRole_shouldThrowSQLException_whenCantCreate() throws SQLException {
+        CreateJobRoleRequest jobRoleRequest = new CreateJobRoleRequest("jobRoleName", "jobSpecSummary",
+                1, 1,
+                "jobResponsibility", "sharepointLink");
+
+        Connection connectionMock = mock(Connection.class);
+        Mockito.when(databaseConnector.getConnection()).thenThrow(SQLException.class);
+        PreparedStatement mockPreparedStatement = mock(PreparedStatement.class);
+        Mockito.when(connectionMock.prepareStatement(any(String.class))).thenReturn(mockPreparedStatement);
+        Mockito.when(mockPreparedStatement.executeUpdate()).thenReturn(-1);
+        assertThrows(SQLException.class,
+                () -> jobRoleDao.createJobRole(jobRoleRequest));
+    }
 }
```

Comment:
Missing test case when -1 is returned from the DAO

Reply:
added